src/Imagine/Cache/Resolver/FormatExtensionResolver.php

```diff
@@ -35,37 +29,28 @@ public function __construct(ResolverInterface $resolver, FilterConfiguration $fi
     /**
      * {@inheritdoc}
      */
-    public function resolve($path, $filter)
+    public function resolve(string $path, string $filter): string
     {
         $path = $this->replaceExtension($path, $filter);
 
         return $this->resolver->resolve($path, $filter);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function store(BinaryInterface $binary, $targetPath, $filter)
+    public function store(BinaryInterface $binary, string $path, string $filter): void
     {
-        $targetPath = $this->replaceExtension($targetPath, $filter);
+        $path = $this->replaceExtension($path, $filter);
 
-        return $this->resolver->store($binary, $targetPath, $filter);
+        $this->resolver->store($binary, $path, $filter);
```

Comment:
the return value doesn't seem to be used, only some of the resolvers were returning a value, the same for `ResolverInterface::remove`.

Reply:
what kind of value returned those that returned anything?

Reply:
so in general, these methods are called from:

https://github.com/liip/LiipImagineBundle/blob/1fb3dd3dd6f99a9fdfc906189f59cbb821e4a215/src/Imagine/Cache/CacheManager.php#L219-L222

and

https://github.com/liip/LiipImagineBundle/blob/1fb3dd3dd6f99a9fdfc906189f59cbb821e4a215/src/Imagine/Cache/CacheManager.php#L253-L255

The ones that were returning values (`PsrCacheResolver`, `ProxyResolver` and `FormatExtensionResolver`) receive a `ResolverInterface` value in the constructor, so they were returning the value from the inner resolver.

Reply:
ah i see. yeah then likely it is all because of no typing and not noticing that we never actually return anything.